### PR TITLE
Fix/vtn 7571 veritone app.login should return a promise

### PR DESCRIPTION
**Pull request by mclenithan**

### CODE REVIEW

File: packages/veritone-redux-common/src/helpers/redux/promiseMiddleware.js [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        pendingActionList.push(newPendingActionInfo);
+        promisesList.push(promise);
+
+        // Move on, but return a promise for dispatch
+        next(action);
+        return promise;
+      };
+    };
+  };
+  return Object.assign(middleware, { getPromisesList });
```
Comment (olslash) [inline packages/veritone-redux-common/src/helpers/redux/promiseMiddleware.js]: promisesList/getPromisesList seems to be useless?

File: packages/veritone-redux-common/src/helpers/redux/promiseMiddleware.test.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    };
+    expect(promWrap(action)).toBeInstanceOf(Promise);
+  });
+
+  it('should resolve promise on successful action', () => {
+    action = {
+      type: 'TEST',
+      [WAIT_FOR_ACTION]: 'TEST_SUCCESS'
+    };
+    promWrap(action).then(result => {
```
Comment (olslash) [inline packages/veritone-redux-common/src/helpers/redux/promiseMiddleware.test.js]: you need to return this promise from the it(), or provide a done arg to it() and call it once async tests are done-- right now this isnt doing anything

https://facebook.github.io/jest/docs/en/asynchronous.html

File: packages/veritone-redux-common/src/helpers/redux/promiseMiddleware.test.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  });
+
+  it('should reject promise on error action and grab payload', () => {
+    action = {
+      type: 'TEST',
+      [WAIT_FOR_ACTION]: 'TEST_SUCCESS',
+      [ERROR_ACTION]: 'TEST_ERROR',
+      [CALLBACK_ERROR_ARGUMENT]: action => action.payload
+    };
+    promWrap(action).catch(result => {
```
Comment (olslash) [inline packages/veritone-redux-common/src/helpers/redux/promiseMiddleware.test.js]: same comment re: async tests

File: packages/veritone-redux-common/src/helpers/redux/promiseMiddleware.js (hunk 1 above)
Comment (mclenithan) [reply packages/veritone-redux-common/src/helpers/redux/promiseMiddleware.js]: unless you want the list of promises...  yea it's useless

File: packages/veritone-redux-common/src/helpers/redux/promiseMiddleware.test.js (hunk 2 above)
Comment (mclenithan) [reply packages/veritone-redux-common/src/helpers/redux/promiseMiddleware.test.js]: good call, I'll change that up

File: packages/veritone-redux-common/src/helpers/redux/promiseMiddleware.test.js [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+  it('should resolve promise on successful action', () => {
+    expect.assertions(2);
+    action = {
+      type: 'TEST',
+      [WAIT_FOR_ACTION]: 'TEST_SUCCESS'
+    };
+    const prom = promWrap(action);
+    expect(promWrap(successAction)).toBeUndefined();
+    return expect(prom).resolves.toBe('winnerwinner');
```
Comment (olslash) [inline packages/veritone-redux-common/src/helpers/redux/promiseMiddleware.test.js]: cool

**Pull request merged**